### DISCUSSION TITLE
Insert current search as a Markdown link

### DISCUSSION THREAD

**Discussion by luckman212** [SilentVoid13/Templater#141] [Templates Showcase]
Will take your current search query and insert a ready-to-go Obsidian link to quickly recall it later. Useful for MOCs.

```js
<%*
  var searchLeaf;
  app.workspace.iterateAllLeaves(leaf => {
  if (leaf.view.getViewType() === 'search') {
    searchLeaf = leaf;
  }});
  if (typeof(searchLeaf) !== 'undefined') {
    var q = searchLeaf.view.getQuery();
    if (q !== '') {
      var md = '[search: ' + q + '](obsidian://search?query=' + encodeURIComponent(q) + ')';
      tR += md;
    } else {
      new Notice('Search field is empty!');
    }
  }
%>
```

**Comment by FelipeRearden** [comment]
Hello @luckman212 !!!!

Thank you very much for sharing this script. 

I would like to ask your help for a problem that I am facing in a script that is related to this script → **instead of getting** `view.getQuery()` I would like to get the search matches. Moreover I need to get the `.basename` of these matches. 

![1B9E3762-CD3A-4BDD-BA6D-51E6F20827F5](https://user-images.githubusercontent.com/85180696/228058993-e4f9242f-4d47-4f12-8651-f5c968eada2f.jpeg)

--- 

Do you have any idea on how to get this ? 

Thanks for your help 🙏

**Comment by luckman212** [comment]
@FelipeRearden Not sure if that's possible. I looked in `app.js` and I found there's some code to copy search results, but not sure yet how to invoke it as the function names are all obfuscated. If I figure it out, I'll post back.

In the meantime, you are aware of this button, yes?

![image](https://user-images.githubusercontent.com/1992842/228107868-e5856b6f-9f25-4e56-a55d-51a2f3fbbb7e.png)

Also, you can [embed search results](https://help.obsidian.md/Plugins/Search#Embed+search+results+in+a+note) directly in your notes:

```md
```query
test-search
```              <---

**Reply by FelipeRearden** [reply]
Thank you very much fro the reply @luckman212 !!!!!!

> @FelipeRearden Not sure if that's possible. I looked in `app.js` and I found there's some code to copy search results, but not sure yet how to invoke it as the function names are all obfuscated. If I figure it out, I'll post back.

I really appreciate your help and trying to find a way to get the function names 🙏🙏🙏

I found this function `search.view.dom.resultDomLookup` on line 10 of this script from our community friend Murf:

→ https://gist.github.com/GitMurf/9e69f456d8a185063ce4a192bafcbfec

Maybe this info could help when looking at `app.js`

> 
> In the meantime, you are aware of this button, yes?
>
> Also, you can [embed search results](https://help.obsidian.md/Plugins/Search#Embed+search+results+in+a+note) directly in your notes:

I really appreciate for these suggestions !

My issue is that I need to use the search result from the search core plugin in a templater script to get the filename of a query like this `alias: [tp.file.selection()]`. This filename will be used as input to this in the rest of the script.

I am trying to use the search query because I cant get the `basename` based on the alias of a note. My idea is to use the search query to get the file where that has the alias that I am searching.

This way, I need to get the search match automatically :)

**Comment by luckman212** [comment]
Hmm. Ok, does this help?

```ts
<%*
const searchView = app.workspace.getLeavesOfType('search')[0]?.view;
if (!searchView) {
    new Notice('Search pane not found', 5000);
    return;
}
const searchTerms = searchView.getQuery();
const searchResults = searchView.dom.getFiles();
if (!searchResults.length) {
    new Notice('No search results available', 5000);
    return;
}
tR += `Results from search query: \`${searchTerms}\`:\n\n`;
searchResults.forEach(res => {
    const bn = res.basename;
    const pth = encodeURIComponent(res.path);
    tR += `- [${bn}](${pth})\n`;
});
%>
```

If you want Wikilinks instead, you can use

```ts
tR += `- [[${bn}]]\n`;
```

**Reply by FelipeRearden** [reply]
First of all, forgive for my late reply. It was past midnight here in my country and I literally felt sleep 🥱

> Hmm. Ok, does this help?

IT IS PERFECT !!!!!!!!!!!!!!!!!!!!!!!!!!!

Thank you very much @luckman212 for your help and dedication with my issue !!!! I have no words to say to you how happy I am :)

Thanks for your saving my Obsidian life 🛟 you are super 🥇

I wish you a fantastic day ☀️
